packages/sitecore-jss-nextjs/src/revalidate/revalidate-middleware.ts

```diff
@@ -0,0 +1,214 @@
+﻿import { NextApiResponse, NextApiRequest } from 'next';
+import { GraphQLPersonalizeService } from '@sitecore-jss/sitecore-jss/personalize';
+import { GraphQLRequestClientFactory } from '@sitecore-jss/sitecore-jss/graphql';
+
+export type Entity = {
+  identifier: string;
+  entity_definition: string;
+  operation: string;
+  entity_culture: string;
+};
+
+export type UpdatedPaths = {
+  path: string;
+  language: string;
+};
+
+export type PersonalizedResult = {
+  path: string;
+  variantId: string;
+};
+
+export type RevalidateConfig = {
+  clientFactory: GraphQLRequestClientFactory;
+  multiSite?: boolean;
+  personalize?: boolean;
+};
+
+const SITE_PREFIX = '_site_';
```

Comment:
We shouldn't be manually constructing these rewrite segments for site and personalize. We have util functions that handle this:
https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss/src/site/utils.ts#L13
https://github.com/Sitecore/jss/blob/dev/packages/sitecore-jss/src/personalize/utils.ts#L14

Reply:
Well I looked at them and these functions work slightly different than what we need. The data I have already has the `siteName` in the path whereas the util functions accepts them separately.

Reply:
I tried to figure out, but I see that you construct the rewrite path manually here: https://github.com/Sitecore/jss/pull/1674/files#diff-1a3c56f1c4214f67a0fa03b77abb29d41607f92bd5d5b2b781e4b09cdef70f90R156
In general it's not really good to duplicate this configuration here, you can try to prepare your parameters in a middleware and pass it to utils in appropriate format

Reply:
I have refactored to use our existing util functions.
Do we have a util function to construct the path with both multiSite and personalize prefixes?